### PR TITLE
perf: remove reporter logs for collapsed tests in run mode

### LINKED ISSUE #25230: Release command log items in run mode
### What would you like?

In headless run mode, the command log cannot be interacted with, as such, the logs in the command log do not need to be retained and can be released. See this [comment](https://github.com/cypress-io/cypress/issues/23391#issuecomment-1357904950) for additional details.

### Why is this needed?

Don't unnecessarily consume memory.

### Other

_No response_

### PR DESCRIPTION
**Pull request by mschile**
- Closes #25230

### Additional details

In order to reduce memory consumption, I am clearing the reporter logs for collapsed tests in run mode.

### Steps to test

Run a spec in `headed` `run` mode and verify a successful test has its reporter logs cleared and a failed test has its reporter logs retained.

### How has the user experience changed?

Before (logs retained):
Screen Shot 2023-01-30 at 2 13 34 PM

After (logs cleared):
Screen Shot 2023-01-30 at 2 15 27 PM

---

Running the following spec produced the following results:
```
describe('memory spec', { browser: { family: 'chromium' } }, () => {
  const text = 'x'.repeat(100000)

  for (let index = 0; index < 50; index++) {
    it(`test ${index + 1} passes`, () => {
      cy.visit('http://localhost:3500/memory')

      for (let index = 0; index < 100; index++) {
        cy.get(`#p${index}`).should('have.text', text)
      }
    })
  }
})
```

```
<html>
  <body></body>
  <script>
    for (let i = 0; i < 100; i++) {
      const el = document.createElement('p')
      el.id = 'p' + i
      el.innerHTML = 'x'.repeat(100000)
            
      document.body.appendChild(el)
    }
  </script>
</html>  
```

Before:
Screen Shot 2023-01-30 at 2 07 59 PM

After:
Screen Shot 2023-01-30 at 2 08 27 PM

### PR Tasks

- [x] Have tests been added/updated?
- [x] Has the original issue (or this PR, if no issue exists) been tagged with a release in ZenHub? (user-facing changes only)
- [n/a] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [n/a] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?

### CODE REVIEW

**Comment by emilyrohrbough** [comment]
@mschile will this retrain the logs of failed tests? wondering if users can still debug when running
```
cypress run --spec record.cy.js --headed --no-exit
```


EDIT: NEVER MIND. I only looked at the pictures and didn't read the full text. You say the logs for the failed tests are retrained.

**Comment by mschile** [comment]
> @mschile will this retrain the logs of failed tests? wondering if users can still debug when running

Yes, when running in headed run mode, the logs will be retained for the failed tests but would be cleared for the successful tests.

File: cli/CHANGELOG.md [hunk 1]
```diff
@@ -7,6 +7,8 @@ _Released 01/31/2023 (PENDING)_
 
 - Fixed an issue where alternative Microsoft Edge Beta, Canary, and Dev binary versions were not being discovered by Cypress. 
   Fixes [#25455](https://github.com/cypress-io/cypress/issues/25455).
+- Improved memory consumption in `run` mode by removing reporter logs for successful tests.
```
Comment (emilyrohrbough) [inline cli/CHANGELOG.md:17]: This should be under the `**Performance:**` changelog section

The semantic PR title prefix should be `perf` as well.

File: cli/CHANGELOG.md [hunk 2]
```diff
@@ -7,6 +7,8 @@ _Released 01/31/2023 (PENDING)_
 
 - Fixed an issue where alternative Microsoft Edge Beta, Canary, and Dev binary versions were not being discovered by Cypress. 
   Fixes [#25455](https://github.com/cypress-io/cypress/issues/25455).
+- Improved memory consumption in `run` mode by removing reporter logs for successful tests. 
+  Fixes [#25230]https://github.com/cypress-io/cypress/issues/25230
```
Comment (emilyrohrbough) [inline cli/CHANGELOG.md]:
```suggestion
  Fixes [#25230](https://github.com/cypress-io/cypress/issues/25230).
```

File: packages/reporter/src/attempts/attempt-model.ts [hunk 3]
```diff
@@ -194,9 +194,23 @@ export default class Attempt {
     }
   }
 
-  @action finish (props: UpdatableTestProps) {
+  @action finish (props: UpdatableTestProps, isInteractive: boolean) {
     this.update(props)
     this.isActive = false
+
+    // if the test is not open and we aren't in interactive mode, clear out the attempt details
+    if (!this.test.isOpen && !isInteractive) {
```
Comment (emilyrohrbough) [inline packages/reporter/src/attempts/attempt-model.ts:202]: Should this be verifying if the attempt was successful before clearing?

Comment (mschile) [reply packages/reporter/src/attempts/attempt-model.ts:202]: Those checks are made in [test.isOpen](https://github.com/cypress-io/cypress/blob/53db0ea71f756cec93efd8cd0c95218aac9b893a/packages/reporter/src/test/test-model.ts#L81).

File: packages/driver/src/cypress.ts [hunk 4]
```diff
@@ -551,7 +551,7 @@ class $Cypress {
 
         // this event is how the reporter knows how to display
         // stats and runnable properties such as errors
-        this.emit('test:after:run', ...args)
+        this.emit('test:after:run', args[0], this.config('isInteractive'))
```
Comment (emilyrohrbough) [inline packages/driver/src/cypress.ts:554]: Was there intentions to release these logs in the [runner.js logic ](https://github.com/cypress-io/cypress/blob/develop/packages/driver/src/cypress/runner.ts#L1738) as well?

We are storing these log values twice.

Comment (mschile) [reply packages/driver/src/cypress.ts:554]: We don't retain the logs in run mode in the runner: https://github.com/cypress-io/cypress/blob/53db0ea71f756cec93efd8cd0c95218aac9b893a/packages/driver/src/cypress/runner.ts#L1761-L1765

**Review by emilyrohrbough — APPROVED** [review]

**Review by emilyrohrbough — APPROVED** [review]
Looks good with changelog update!

**Review by mjhenkes — APPROVED** [review]

File: cli/CHANGELOG.md [hunk 5]
```diff
@@ -8,6 +8,10 @@ _Released 01/31/2023 (PENDING)_
 - Fixed an issue where alternative Microsoft Edge Beta, Canary, and Dev binary versions were not being discovered by Cypress. 
   Fixes [#25455](https://github.com/cypress-io/cypress/issues/25455).
 
+**Performance:**
+- Improved memory consumption in `run` mode by removing reporter logs for successful tests. 
+  Fixes [#25230]https://github.com/cypress-io/cypress/issues/25230.
```
Comment (emilyrohrbough) [inline cli/CHANGELOG.md]:
```suggestion
  Fixes [#25230](https://github.com/cypress-io/cypress/issues/25230).
```

**Review by AtofStryker — APPROVED** [review]

**Pull request merged**